WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -1146,20 +1146,19 @@ private extension ProductFormViewController {
             return
         }
 
-        let downloadFileListViewController = ProductDownloadListViewController(product: product) { [weak self] data in
-            self?.onAddEditDownloadsCompletion(data: data)
+        let downloadFileListViewController = ProductDownloadListViewController(product: product) { [weak self] (data, hasUnsavedChanges) in
```

Comment:
It's not clear to me why the `ProductDownloadListViewController` is returning if there are unsaved changes. If there are unsaved changes, you should just compare the original data and the new data returned.

Reply:
Updated